framework/src/kernels/ADKernelGrad.C

```diff
@@ -0,0 +1,148 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ADKernelGrad.h"
+#include "Assembly.h"
```

Comment:
I know this is copy-pasta but some of these includes appear unnecessary and I needed to at least one other review comment